### PR TITLE
adds page titles to each page

### PR DESCRIPTION
**Pull request by jason00111**

### CODE REVIEW

File: web-server/helpers.js [hunk 1]
```diff
@@ -167,3 +170,7 @@ module.exports = app => {
   })
 
 }
+
+function getTitleFromHTML(content) {
+  return content.match(/>(.*)</)[1]
```
Comment (deadlyicon) [inline web-server/helpers.js]: this is sneaky. How does this work?

File: web-server/views/layout.jade [hunk 2]
```diff
@@ -4,7 +4,7 @@ html(lang='en')
     meta(charset='utf-8')
     meta(http-equiv="X-UA-Compatible" content="IE=edge,chrome=1")
     meta(name="viewport" content="width=device-width, initial-scale=1")
-    title Curriculum - Learners Guild
+    title= title
```
Comment (deadlyicon) [inline web-server/views/layout.jade]: a default and a suffix would be nice.

```pug
title
  = title ? `${title} | `
  Curriculum - Learners Guild
```

File: web-server/helpers.js (hunk 1 above)
Comment (jason00111) [reply web-server/helpers.js]: `content` is a string of HTML which is the result of parsing markdown.
Line 175 looks in this string for the contents of the first html tag which is the title.
We'll update it to look for the first `h1` tag to make it more robust.

File: web-server/helpers.js [hunk 3]
```diff
@@ -103,7 +103,7 @@ module.exports = app => {
             if (files.includes('README.md')){
               response.renderMarkdownFile(relativePath+'/README.md')
             }else{
-              response.render('directory', {files})
+              response.render('directory', {files, title: 'Directory'})
```
Comment (deadlyicon) [inline web-server/helpers.js]: maybe this should be or at least include the `relativePath` value?

File: web-server/helpers.js [hunk 4]
```diff
@@ -187,3 +190,7 @@ module.exports = app => {
   })
 
 }
+
+function getTitleFromHTML(content) {
+  return content.match(/<h1.*>(.*)<\/h1>/)[1]
```
Comment (deadlyicon) [inline web-server/helpers.js]: This might be a bit more accurate of a regexp `/<h1[^>]*>([^<]*)<\/h1>/`

File: web-server/views/layout.jade [hunk 5]
```diff
@@ -4,7 +4,7 @@ html(lang='en')
     meta(charset='utf-8')
     meta(http-equiv="X-UA-Compatible" content="IE=edge,chrome=1")
     meta(name="viewport" content="width=device-width, initial-scale=1")
-    title Curriculum - Learners Guild
+    title= (title ? `${title} | ` : '') + 'Curriculum - Learners Guild'
```
Comment (deadlyicon) [inline web-server/views/layout.jade]: 👍

**Pull request merged**